### PR TITLE
ffi: cleanup setobject

### PR DESCRIPTION
**Pull request by deantvv**
For #1289

- Change the order of declaration as cpython does
- Document skipped function/macro

### CODE REVIEW

File: src/ffi/setobject.rs [hunk 1]
```diff
@@ -46,12 +29,78 @@ pub struct PySetObject {
...
+    #[cfg_attr(PyPy, link_name = "PyPySet_Discard")]
+    pub fn PySet_Discard(set: *mut PyObject, key: *mut PyObject) -> c_int;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Pop")]
+    pub fn PySet_Pop(set: *mut PyObject) -> *mut PyObject;
+    #[cfg_attr(PyPy, link_name = "PyPySet_Size")]
+    pub fn PySet_Size(anyset: *mut PyObject) -> Py_ssize_t;
+}
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
+extern "C" {
```
Comment (davidhewitt) [inline src/ffi/setobject.rs:79]: Usually the `windows` link name attribute is only needed for `static mut` symbols, you might be able to merge this `extern` block with the one above:

```suggestion
```

File: src/ffi/setobject.rs [hunk 2]
```diff
@@ -46,12 +29,78 @@ pub struct PySetObject {
...
 #[cfg(not(PyPy))]
 pub unsafe fn PyFrozenSet_CheckExact(ob: *mut PyObject) -> c_int {
     (Py_TYPE(ob) == &mut PyFrozenSet_Type) as c_int
 }
 
+#[inline]
+pub unsafe fn PyFrozenSet_Check(ob: *mut PyObject) -> c_int {
+    (Py_TYPE(ob) == &mut PyFrozenSet_Type
+        || PyType_IsSubtype(Py_TYPE(ob), &mut PyFrozenSet_Type) != 0) as c_int
+}
```
Comment (davidhewitt) [inline src/ffi/setobject.rs:95]: This one should also be `not(PyPy)` like `PyFrozenSet_CheckExact`  - I see a `PyPyFrozenSet_Check` in my list of symbols from last time I extracted the PyPy lib.

File: src/ffi/setobject.rs [hunk 3]
```diff
@@ -65,42 +114,17 @@ pub unsafe fn PyAnySet_Check(ob: *mut PyObject) -> c_int {
         || PyType_IsSubtype(Py_TYPE(ob), &mut PyFrozenSet_Type) != 0) as c_int
 }
 
+// skipped PySet_CheckExact
+
+#[cfg_attr(windows, link(name = "pythonXY"))]
```
Comment (davidhewitt) [inline src/ffi/setobject.rs:119]: I think no need for this attribute (the imported symbol is a function, not `static mut`)

```suggestion
```

**Review by davidhewitt — COMMENTED** [review]
Thanks again - looks great! Just a few final pieces I spotted when cross-checking...

File: src/ffi/setobject.rs (hunk 1 above)
Comment (deantvv) [reply src/ffi/setobject.rs:79]: > Usually the `windows` link name attribute is only needed for `static mut` symbols
Could you please elaborate why?
In my understanding, `link` statement will be emitted in build script. Why this is necessary to add the attribute only to `static mut` symbols but not function?
Thanks in advance. I couldn't understand this after researching for several hours.

>, you might be able to merge this `extern` block with the one above:
Will do.

File: src/ffi/setobject.rs (hunk 3 above)
Comment (deantvv) [reply src/ffi/setobject.rs:119]: Same as above.

File: src/ffi/setobject.rs (hunk 2 above)
Comment (deantvv) [reply src/ffi/setobject.rs:95]: Thanks for catching this.

**Review by davidhewitt — APPROVED** [review]
Perfect, thanks again!

File: src/ffi/setobject.rs (hunk 1 above)
Comment (davidhewitt) [reply src/ffi/setobject.rs]: Yeah, unfortunately I haven't found this documented anywhere. I suspect it's not intended, however in practice I've found that we experience windows linker errors for `static mut` symbols if they aren't covered explicitly by a `link_name` attribute. I should really report this upstream...

**Pull request merged**